demisto_sdk/commands/format/update_layout.py

```diff
@@ -255,3 +261,83 @@ def remove_copy_and_dev_suffixes_from_layout(self):
                     if section.get('queryType') == SCRIPT_QUERY_TYPE:
                         section['query'] = remove_copy_and_dev_suffixes_from_str(section.get('query'))
                         section['name'] = remove_copy_and_dev_suffixes_from_str(section.get('name'))
+
+    def remove_non_existent_fields_container_layout(self):
+        """
+        Remove non-existent fields from a container layout.
+        """
+        if not self.id_set_file:
+            return
+
+        content_fields = self.get_available_content_fields()
+
+        layout_container_items = [
+            layout_container_field for layout_container_field in LAYOUT_CONTAINER_FIELDS
+            if self.data.get(layout_container_field)
+        ]
+
+        for layout_container_item in layout_container_items:
+            layout = self.data.get(layout_container_item, {})
+            layout_tabs = layout.get('tabs', [])
+            self.remove_non_existent_fields_from_tabs(
+                layout_tabs=layout_tabs, content_fields=content_fields
+            )
+
+    def remove_non_existent_fields_layout(self):
+        """
+        Remove non-existent fields from a layout.
+        """
+        if not self.id_set_file:
+            return
+
+        content_fields = self.get_available_content_fields()
+
+        layout = self.data.get('layout', {})
+        layout_sections = layout.get('sections', [])
+        for section in layout_sections:
+            fields = section.get('fields', [])
+            section['fields'] = self.extract_content_fields(fields=fields, content_fields=content_fields)
+
+        self.remove_non_existent_fields_from_tabs(
+            layout_tabs=layout.get('tabs', []), content_fields=content_fields
+        )
+
+    def get_available_content_fields(self):
+        """
+        Get all the available content indicator/incident fields available + all the built in fields.
+        """
+        return get_all_incident_and_indicator_fields_from_id_set(self.id_set_file, 'layout') + [
+            field.lower() for field in BUILT_IN_FIELDS
+        ] + LAYOUT_AND_MAPPER_BUILT_IN_FIELDS
+
+    @staticmethod
+    def extract_content_fields(fields: list, content_fields: list) -> list:
+        """
+        Get only incident/indicator fields which are part of the id json file.
+
+        Args:
+            fields (list[dict]): list of fields.
+            content_fields (list[str]): all the available content fields from id json.
+
+        Returns:
+            list: fields which are part of the content items (id json).
+        """
+        return [
+            field for field in fields if
+            field.get('fieldId', '').replace('incident_', '').replace('indicator_', '').lower() not in
```

Comment:
lets make a normalize_field_name method and add to tools, since we're doing this twice

Reply:
done